src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -160,7 +160,9 @@ public override void VisitProperty(IPropertySymbol symbol)
                 AddPunctuation(SyntaxKind.OpenBraceToken);
 
                 AddAccessor(symbol, symbol.GetMethod, SyntaxKind.GetKeyword);
-                AddAccessor(symbol, symbol.SetMethod, SyntaxKind.SetKeyword);
+                // PROTOTYPE(init-only): adjust SymbolDisplayVisitor once we have a public IsInitOnly API
+                var keywordForSetAccessor = ((Symbols.PublicModel.MethodSymbol)symbol.SetMethod)?.UnderlyingMethodSymbol.IsInitOnly == true ? SyntaxKind.InitKeyword : SyntaxKind.SetKeyword;
```

Comment:
>(Symbols.PublicModel.MethodSymbol) [](start = 45, length = 34)

This cast is not safe, we could be dealing with VB symbol, for example. We should do a type check instead.